src/typechecker/derive-type.lisp

```diff
@@ -548,6 +548,16 @@ EXPL-DECLARATIONS is a HASH-TABLE from SYMBOL to SCHEME"
                      (some (lambda (b) (not (coalton-impl/ast::node-abstraction-p (cdr b))))
                            bindings)))
 
+            ;; NOTE: This is where defaulting happens
```

Comment:
could you explain a little more in this comment how defaulting relates to the code below?